app/controllers/auth_key_pair_cloud_controller.rb

```diff
@@ -79,16 +68,16 @@ def set_form_vars
     session[:edit] = @edit
   end
 
-  def form_field_changed
-    return unless load_edit("auth_key_pair_cloud_edit__#{params[:id] || 'new'}")
-    get_form_vars
-    enable_buttons = (!@edit[:new][:name].blank? && !@edit[:new][:ems_id].blank?)
-
-    render :update do |page|
-      page << javascript_prologue
-      page.replace(@refresh_div, :partial => @refresh_partial) if @refresh_div
-      page << javascript_for_miq_button_visibility(enable_buttons)
+  # REST call for provider choices
+  def ems_form_choices
+    assert_privileges("auth_key_pair_cloud_new")
+    ems_choices = []
+    ems_choices = ManageIQ::Providers::CloudManager.select { |ems| ems.class::AuthKeyPair.is_available?(:create_key_pair, ems) }.map do |ems|
```

Comment:
probably use `do`.. `end` here for both blocks
on shorter terminla this looks strange and unclear